app/src/main/java/com/globant/counter/mvp/view/CountView.kt

```diff
@@ -2,25 +2,63 @@ package com.globant.counter.mvp.view
 
 
 import android.app.Activity
+import android.text.TextUtils
+import com.globant.counter.mvp.model.CalculatorItem
 import com.globant.counter.utils.bus.RxBus
-import com.globant.counter.utils.bus.observer.OnCountButtonPressedBusObserver
-import com.globant.counter.utils.bus.observer.OnResetButtonPressedBusObserver
+import com.globant.counter.utils.bus.observer.OnCalculatorActionItemPressedBusObserver
+import com.globant.counter.utils.bus.observer.OnCalculatorClearButtonPressedBusObserver
+import com.globant.counter.utils.bus.observer.OnCalculatorEqualButtonPressedBusObserver
+import com.globant.counter.utils.bus.observer.OnCalculatorNumberButtonPressedBusObserver
 import kotlinx.android.synthetic.main.activity_main.*
 
 
 class CountView(activity: Activity) : ActivityView(activity) {
+    fun setExpression(expression: String) {
+        activity?.textExpression?.text = expression
+    }
 
     init {
-        activity.count_button.setOnClickListener {
-            RxBus.post(OnCountButtonPressedBusObserver.OnCountButtonPressed())
+        activity.btnSum.setOnClickListener {
+            postOperation(CalculatorItem.ADDITION)
+        }
+
+        activity.btnDiv.setOnClickListener {
+            postOperation(CalculatorItem.DIVISION)
+        }
+
+        activity.btnMulti.setOnClickListener {
+            postOperation(CalculatorItem.MULTIPLICATION)
+        }
+
+        activity.btnRest.setOnClickListener {
+            postOperation(CalculatorItem.SUBSTRACTION)
         }
 
-        activity.reset_button.setOnClickListener {
-            RxBus.post(OnResetButtonPressedBusObserver.OnResetButtonPressed())
+        activity.btnEqual.setOnClickListener {
+            getInput()
+            RxBus.post(OnCalculatorEqualButtonPressedBusObserver.OnCalculatorEqualButtonPressed())
         }
+
+        activity.btnClear.setOnClickListener {
+            RxBus.post(OnCalculatorClearButtonPressedBusObserver.OnCalculatorClearButton())
+        }
+
     }
 
-    fun setCount(count: String) {
-        activity!!.count_label.text = count
+    private fun getInput() {
+        var input = activity?.inputNumber?.text.toString()
+        if (!input.isNullOrEmpty()) {
+            RxBus.post(OnCalculatorNumberButtonPressedBusObserver.OnCalculatorNumberButtonPressed(input.toFloat()))
+        }
+        activity?.inputNumber?.setText("")
```

Comment:
@taidy-marrugo this is a "magic number". As a rule in royal project, it is enforced that if you create a constant for values that you want to set or compare to variables. In the project we already have an empty_string constant but please create it in your calculator project. so you reuse it and don't go creating the same value over and over again. You use the empty string also in the lines bellow so abstract it to a constant.